app/_src/gateway/production/benchmark.md

```diff
@@ -0,0 +1,218 @@
+
+If you use Redis and any plugin is enabled, ensure Redis is not a bottleneck.
+The CPU generally creates a bottleneck for Redis, so check CPU usage first.
+If this is the case, scale Redis vertically by giving it an additional CPU.
+
+#### DNS
+
+**Action:** Increase `dns_stale_ttl` to `300` or up to `86400`.
+
+DNS servers can bottleneck {{site.base_gateway}} since {{site.base_gateway}} depends on DNS to determine where to send the request.
+
+In the case of Kubernetes, DNS TTLs are five seconds long and can cause problems.
+You can increase `dns_stale_ttl` to `300` or up to `86400` to rule out DNS as the issue.
+
+If DNS servers are the root cause, you will see `coredns` pods creating a bottleneck on the CPU.
+
+### Blocking I/O for access logs
+
+**Action:** Disable access logs for high throughput benchmarking tests by setting the `proxy_access_log` configuration parameter to `off`.
+
+{{site.base_gateway}} and the underlying NGINX are programmed for non-blocking network I/O and they avoid blocking disk I/O as much as possible. However, access logs are enabled by default, and if the disk powering a {{site.base_gateway}} node is slow for any reason, it can result in performance loss.
+Disable access logs for high throughput benchmarking tests by setting the `proxy_access_log` configuration parameter to `off`.
+
+### Internal errors in {{site.base_gateway}}
+
+**Action:** Make sure that there are no errors in {{site.base_gateway}}’s error log.
+
+Check {{site.base_gateway}}’s error log for internal errors. Internal errors can highlight issues within {{site.base_gateway}} or a third-party system that {{site.base_gateway}} relies on to proxy traffic.
+
+## Example kong.conf for benchmarking
+
+The following `kong.conf` file examples contain all the recommended parameters from the previous section:
+
+### Configuration values
+```bash
+# change this for Kubernetes setup to a number matching the CPU limit. Suggested value is 4 or 8.
+nginx_worker_processes=auto
+
+upstream_keepalive_max_requests=100000
+nginx_http_keepalive_requests=100000
+
+proxy_access_log=off
+
+dns_stale_ttl=3600
+```
+
+### Environment variable format
+
+```bash
+# Change this for Kubernetes setup to a number matching CPU limit. We recommend four or eight.
+KONG_NGINX_WORKER_PROCESSES="auto"
+KONG_UPSTREAM_KEEPALIVE_MAX_REQUESTS="100000"
+KONG_NGINX_HTTP_KEEPALIVE_REQUESTS="100000"
+
+KONG_PROXY_ACCESS_LOG="off"
+
+KONG_DNS_STALE_TTL="3600"
+```
+
+### Configuration for Helm chart values.yaml
+
+```bash
+# The value of four for worker process is a suggested value. You can use eight as well.
+# Allocate the same amount of CPU and appropriate memory to avoid OOM killer.
+env:
+  nginx_worker_processes: "4"
+  upstream_keepalive_max_requests: "100000"
+  nginx_http_keepalive_requests: "100000"
+  proxy_access_log: "off"
+  dns_stale_ttl: "3600"
+
+resources:
+  requests:
+    cpu: 4
+    memory: ""
+  limits:
+    cpu: 4
+    memory: ""
+```
```

Comment:
Let's move these into tabs instead of headings: kong.conf, environment variables, and Helm chart values